packages/flutter/lib/src/services/platform_views.dart

```diff
@@ -503,14 +504,18 @@ class _AndroidMotionEventConverter {
     _pointTransformer = transformer;
   }
 
-  int nextPointerId = 0;
   int downTimeMillis;
 
   void handlePointerDownEvent(PointerDownEvent event) {
-    if (nextPointerId == 0) {
+    if (pointerProperties.isEmpty) {
       downTimeMillis = event.timeStamp.inMilliseconds;
     }
-    pointerProperties[event.pointer] = propertiesFor(event, nextPointerId++);
+    int androidPointerId = 0;
+    while (usedAndroidPointerIds.contains(androidPointerId)) {
```

Comment:
qq: how does this ensure that pointer id doesn't reach the limit? As pointed out in https://github.com/flutter/flutter/issues/61421#issuecomment-659715358

Reply:
The `MotionEvent` pointer ID limit is an internal Android implementation detail that AFAICT isn't documented anywhere.

This patch ensures that the pointer ID value will be less than the number of active pointers.  The version of Android that I looked at apparently accepts pointer IDs in the range 0 through 31.  So realistically this patch will keep the pointer ID values within a safe range given that a user is unlikely to have >32 pointers active simultaneously.

Given that it isn't a defined part of Android's API I didn't want to try to hardcode a limit here.  But we could enforce a maximum number of active pointers on the framework side if this is a concern.